source/simulator/assembly.cc

```diff
@@ -903,6 +922,37 @@ namespace aspect
                                                         advection_field);
       }
 
+    // if set to true, the maximum of the artificial viscosity in the cell as well
+    // as the neighbors of the cell is computed and used instead
+    if (use_smoothing == true)
+      {
+        Vector<T> viscosity_per_cell_temp;
+        viscosity_per_cell_temp.reinit(triangulation.n_active_cells());
+
+        viscosity_per_cell = 0.0;
```

Comment:
I doubt this will work because you are setting the values to zero that you computed above.